Add PWM support for Renesas RA6, RA4, RA2 devices

khoa-nguyen-18 opened the pull request:
Enable PWM support on these boards:
- RA6: ek_ra6e2, ek_ra6m1, ek_ra6m2, ek_ra6m3, ek_ra6m4, ek_ra6m5, fpb_ra6e1, fpb_ra6e2
- RA4: ek_ra4e2, ek_ra4m2, ek_ra4m3, ek_ra4w1
- RA2: ek_ra2a1

Commit:
dts: arm: renesas: add PWM support for Renesas RA6, RA4, RA2

Add PWM support for RA6, RA4, RA2 MCU: ra6-cm4, ra6-cm33,
ra4-cm4, ra4-cm33, ra2xx

Signed-off-by: Danh Doan <danh.doan.ue@bp.renesas.com>

Commit:
drivers: pwm: update namming of pwm driver for RA family

 - update namming for pwm driver.

Signed-off-by: Danh Doan <danh.doan.ue@bp.renesas.com>

Commit:
tests: drivers: pwm: Support RA2, RA4, RA6 to test pwm_loopback

Add overlay to support test pwm_loopback on boards: ek_ra2a1,
ek_ra4e2, ek_ra4m2, ek_ra4m3, ek_ra4w1, ek_ra6e2, ek_ra6m1,
ek_ra6m2, ek_ra6m3, ek_ra6m4, ek_ra6m5, fpb_ra6e1, fpb_ra6e2.

Signed-off-by: Danh Doan <danh.doan.ue@bp.renesas.com>

Commit:
boards: renesas: Add PWM support for Renesas RA6, RA4, RA2

Add pinctrl & node of pwm driver for these board:
ek_ra2a1, ek_ra4e2, ek_ra4m2, ek_ra4m3, ek_ra4w1,
ek_ra6e2, ek_ra6m1, ek_ra6m2, ek_ra6m3, ek_ra6m4,
ek_ra6m5, fpb_ra6e1, fpb_ra6e2.

Signed-off-by: Danh Doan <danh.doan.ue@bp.renesas.com>

khoa-nguyen-18 commented:
Just rebase main to solve twister test fail

soburi commented on dts/bindings/pwm/renesas,ra-pwm.yaml:
```diff
@@ -1,9 +1,9 @@
+description: Renesas RA Pulse Width Modulation
 
-compatible: "renesas,ra8-pwm"
+compatible: "renesas,ra-pwm"
```
This change makes it necessary to update the migration guide.

thaoluonguw reviewed (dismissed):
It looks good to me.

bjarki-andreasen commented:
could you rebase and add an entry in the migration guide regarding the updated naming of ra8 to ra?

khoa-nguyen-18 commented:
Currently the test `pwm_loopback` has an issue. We're hopeful this PR will be merged soon: https://github.com/zephyrproject-rtos/zephyr/pull/82623

khoa-nguyen-18 committed:
doc: releases: Add migration guide to update name for PWM Renesas

Update the naming of Renesas PWM compatible from "renesas,ra8-pwm"
to "renesas,ra-pwm"

Signed-off-by: Khoa Nguyen <khoa.nguyen.xh@renesas.com>

khoa-nguyen-18 commented:
> could you rebase and add an entry in the migration guide regarding the updated naming of ra8 to ra?

We have added a commit to update the naming of Renesas PWM compatible from `renesas,ra8-pwm` to `renesas,ra-pwm`. Thanks you.

bjarki-andreasen reviewed (dismissed):
Nice!
